### PR TITLE
refactor: move a dependency installation to test script

### PR DESCRIPTION
**Pull request by dpigasin**
this fixes tests on versions younger than 2021.2

### CODE REVIEW

File: tests/functional/test_base_cpu.py [hunk 1]
```diff
@@ -20,7 +20,6 @@ def test_base_cpp(tester, image, install_openvino_dependencies):
     }
     tester.test_docker_image(
         image,
-        [install_openvino_dependencies,
```
Comment (generalova-kate) [inline tests/functional/test_base_cpu.py:23]: add mark < 2021.3 to test and create one more test for >= 2021.3

File: tests/resources/base_cpu/demo.sh [hunk 2]
```diff
@@ -2,6 +2,9 @@
 # Copyright (C) 2019-2020 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+apt update
+apt install -y build-essential curl cmake
```
Comment (generalova-kate) [inline tests/resources/base_cpu/demo.sh:6]: maybe add curl, cmake, gcc, g++ to base and custom Dockerfiles. And install only make in the test.

File: tests/functional/test_base_cpu.py (hunk 1 above)
Comment (dpigasin) [reply tests/functional/test_base_cpu.py:23]: what is the difference between them?

File: tests/resources/base_cpu/demo.sh (hunk 2 above)
Comment (dpigasin) [reply tests/resources/base_cpu/demo.sh:6]: OV installs the same packages in test scripts
https://github.com/openvinotoolkit/openvino/tree/master/scripts/demo

File: dockerfiles/ubuntu18/openvino_c_base_2020.1.dockerfile [hunk 3]
```diff
@@ -57,6 +57,15 @@ LABEL Vendor="Intel Corporation"
 COPY --from=base /opt/intel /opt/intel
 RUN echo "source /opt/intel/openvino/bin/setupvars.sh" | tee -a /root/.bashrc
 
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+      curl \
+      ca-certificates \
```
Comment (generalova-kate) [inline dockerfiles/ubuntu18/openvino_c_base_2020.1.dockerfile:63]: ca-certificates, make are under GPL

**Review by generalova-kate — APPROVED** [review]

**Pull request merged**